src/vs/workbench/api/browser/mainThreadEditorTabs.ts

```diff
@@ -199,6 +200,24 @@ export class MainThreadEditorTabs implements MainThreadEditorTabsShape {
 			};
 		}
 
+		if (editor instanceof MultiDiffEditorInput) {
+			const diffEditors: TextDiffInputDto[] = [];
+			for (const resource of (editor?.initialResources ?? [])) {
+				if (resource.original && resource.modified) {
+					diffEditors.push({
+						kind: TabInputKind.TextDiffInput,
+						original: resource.original,
+						modified: resource.modified
+					});
+				}
+			}
+
```

Comment:
Note that the resources of a multi diff editor can change!
For scm, `initialResources` is empty, and the resources are loaded via the multi diff editor source uri.

Reply:
Hmm do you have suggestions on how to tackle this? Is there an event that fires when these resources change? I'm not sure we have other editor inputs that are like this as normally the input just becomes a new one if resources change

Reply:
Yes, I'd suggest to not expose the resources for now.